### ISSUE TITLE
sshcode to a MacOS

### ISSUE THREAD

**Issue by holeybit** [cdr/sshcode#70]
Is `sshcode` supposed to work connecting to a Mac?

```
wget -N https://codesrv-ci.cdr.sh/latest-linux
```

1. looks like it tries to download linux version.
2. `wget` is not available by default. (curl is)

cheers,

**Comment by ammario** [comment]
We haven’t tried to support it yet.

**Comment by sreya92** [comment]
The supported use-case is a remote linux machine, personnaly I tend to test on ubuntu 16.04. However, removing external dependencies (other than ssh) is something we've been considering.

**Comment by holeybit** [comment]
Are you guys open for PR? I've made the following changes to support both linux and Mac `code-server`:

```go
func downloadScript(codeServerPath string) string {
	return fmt.Sprintf(
		`set -euxo pipefail || exit 1
mkdir -p ~/.local/share/code-server %v
cd %v
if [[ "$OSTYPE" == "linux-gnu" ]]; then
  curl -z latest-code-server -o latest-code-server https://codesrv-ci.cdr.sh/latest-linux
elif [[ "$OSTYPE" == "darwin"* ]]; then
  curl -z latest-code-server -o latest-code-server https://codesrv-ci.cdr.sh/latest-darwin
fi
[ -f %v ] && rm %v
ln latest-code-server %v
chmod +x %v`,
		filepath.Dir(codeServerPath),
		filepath.Dir(codeServerPath),
		codeServerPath,
		codeServerPath,
		codeServerPath,
		codeServerPath,
	)
}
```

https://github.com/holeybit/sshcode/tree/feat/support-mac-code-server

Obviously, `https://codesrv-ci.cdr.sh/latest-darwin` does not exist currently. But when I use my own server, I was able to test on ubuntu 16.04, 18.04 and MacOS Mojave.

**Comment by sreya92** [comment]
I think for now we're just going to support remote Linux environments. If it becomes obvious that a lot of people want/need remote Mac environments then we would potentially be open to supporting it...for now the goal is to keep sshcode as simple as possible

**Comment by holeybit** [comment]
Sounds good. I'm currently using this via `ngrok` from my laptop (on the go) to my desktop Mac at home. Would be nice to have an official support in the future. Thanks!

**Comment by sreya92** [comment]
Ah that makes sense, thanks for your interest in the project! Please don't hesitate to file additional issues as you come across them

**Comment by martimarkov** [comment]
@holeybit where did you get the binary for code-server?

**Comment by fabianmu** [comment]
I'd also love to see macos as remote machine

**Comment by nhooyr** [comment]
Yes will reopen, I'm sure people have Mac minis lying around they wanna use.

**Comment by fabianmu** [comment]
Ace, thanks! That's exactly my case - my main machine is a mac mini in the office that I would like to connect to.

**Comment by judewang** [comment]
Me, too. In my case, I really want to replace my 15” MacBook Pro with iPad Pro + iMac.

**Comment by nhooyr** [comment]
Coming soon ;)

**Comment by nhooyr** [comment]
@judewang You wouldn't be able to use sshcode on an iPad though.

**Comment by judewang** [comment]
@nhooyr I forgot... Only thought about I have ssh client on my iPad.

**Comment by fabianmu** [comment]
Any updates on this @nhooyr ?

**Comment by nhooyr** [comment]
See https://github.com/cdr/sshcode/issues/185

Best to use the code-server install script for now and forward the port yourself. See https://github.com/cdr/code-server#quick-install

We plan on adding SSH support to it to replace sshcode. See https://github.com/cdr/code-server/issues/1729

**Comment by fabianmu** [comment]
Oh that works, splendid, thanks @nhooyr!
